### PR TITLE
pass in GoogleCloudStorage required environment variable

### PR DESCRIPTION
**Pull request by mvanholsteijn**
In support of GCS support in PrivateBin.

https://github.com/PrivateBin/PrivateBin/issues/794
https://github.com/PrivateBin/PrivateBin/pull/795

### CODE REVIEW

**Comment by elrido** [comment]
I'll get this merged once the upstream change is in.

**Comment by mvanholsteijn** [comment]
> I'll get this merged once the upstream change is in.

That will be great. I got the Cloud Run deployment of PrivateBin running with these changes.

https://gist.github.com/mvanholsteijn/9f2ad5a3c88f4dcbdabd3bff23180e4a

**Comment by mvanholsteijn** [comment]
@elrido, I saw you added the google/cloud-storage libraries into the image. Can you please merge this change too?

File: etc/php8/php-fpm.d/zz-docker.conf [hunk 1]
```diff
@@ -11,3 +11,9 @@ pm.max_children = 5
 pm.start_servers = 2
 pm.min_spare_servers = 1
 pm.max_spare_servers = 3
+
+env[PRIVATEBIN_GCS_BUCKET] = $PRIVATEBIN_GCS_BUCKET
```
Comment (rugk) [inline etc/php8/php-fpm.d/zz-docker.conf:15]: Is not that passed in by default?

**Review by rugk — APPROVED** [review]

File: etc/php8/php-fpm.d/zz-docker.conf (hunk 1 above)
Comment (elrido) [reply etc/php8/php-fpm.d/zz-docker.conf:15]: It used to be the case that (exported) environment variables got passed down from the web server to (f)cgi scripts/processes along side with HTTP headers and other CGI variables exposed as environment variables. This lead to a number of unexpected side effects[1] and attacks[2] through bugs in the shell that may be in use between web server and the CGI executable and the script languages used in these, themselves. Mitigations were taken on several levels, including the bash shell, webservers and in several scripting languages.

In php-fpm, the mitigation consisted of the introduction of the `clear_env` configuration[3], set to `yes` by default, in release 5.6.0. This got backported to 5.4.27 and 5.5.11, showing how much of a concern this was at the time. The setting is still present in php-fpm 8.0, which is used in our container. Hence the need to explicitly list the environment variables to expose to our app, as we don't want to disable `clear_env`.

[1] https://owasp.org/www-community/attacks/Buffer_Overflow_via_Environment_Variables
[2] https://access.redhat.com/articles/1200223 - see the section "Common Configuration Examples" on how this impacted Apache httpd via the bash shell
[3] https://www.php.net/manual/en/install.fpm.configuration.php#clear-env

**Pull request merged**